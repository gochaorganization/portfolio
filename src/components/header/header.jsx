import React, { Component } from 'react'
import PropTypes from 'prop-types'

import _ from 'lodash'
import classNames from 'classnames'

import './header.sass'

const renderLink = ({ name, link }, key) => (
  <li key={key}>
    <a className="header__list-link" href={link}>
      {name}
    </a>
  </li>
)

renderLink.propTypes = {
  name: PropTypes.string.isRequired,
  link: PropTypes.string.isRequired,
}

class Header extends Component {
  state = {
    menuVisible: false,
    links: [
      { name: 'Blog', link: '#blog' },
      { name: 'Style Guide', link: '#guide' },
      { name: 'Manuals', link: '#manuals' },
      { name: 'Pages', link: '#pages' },
      { name: 'Contacts', link: '#contacts' },
    ],
  }

  toogleMenu = () => {
    this.setState(prevState => ({
      menuVisible: !prevState.menuVisible,
    }))
  }

  render() {
    const { menuVisible, links } = this.state

    return (
      <header className="header">
        <div className="header-wrapper">
          <a href="/" className="header__home">
            Selleo
          </a>
          <span
            role="button"
            tabIndex="0"
            className={classNames('header__menu', 'icon-menu', {
              'header__menu--active': !menuVisible,
            })}
            onClick={this.toogleMenu}
          />
        </div>
        <ul
          className={classNames('header__list', {
            'header__list--hidden': !menuVisible,
          })}
        >
          {_.map(links, (item, key) => renderLink(item, key))}
        </ul>
      </header>
    )
  }
}

export default Header
