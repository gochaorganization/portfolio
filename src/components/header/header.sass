@import '../../application'

.header
    display: flex
    flex-direction: column    

    &__home
        align-self: center    
        color: $black
        font-size: $font-lg
        font-weight: $font-weight-bold    

    &__list
        display: flex
        flex-direction: column
        list-style: none
        padding: 0
        margin: 0
               
        &-link            
            color: $grey
            display: flex
            font-size: $font-xs
            justify-content: space-around
            padding: $m-xs

    &-wrapper
        display: flex
        justify-content: space-between

    &__menu
        &:before
            color: $bright-grey
            cursor: pointer
            font-size: $menu-icon-size

        &--active:before
            color: $black

+media-breakpoint-up(sm)    
    .header
        flex-direction: row
        justify-content: space-between

        &__home
            font-size: $font-base

        &__list
            display: flex
            flex-direction: row

        &__menu
            display: none


+media-breakpoint-down(xs)  
    .header__list
        +visible

        &--hidden
            +invisible        
