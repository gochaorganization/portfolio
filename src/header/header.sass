@import '../application' 
    
.header
    display: flex
    justify-content: space-between
    
    &__list
        display: flex
        list-style: none

        &-link    
            color: $dark-grey
            font-size: $font-xs
            padding: $m-xs
        
    &__link
        align-self: center
        color: $black
        font-size: $font-lg
        font-weight: bold
