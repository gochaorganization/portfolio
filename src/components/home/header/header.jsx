import React, { Component } from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'

import _ from 'lodash'
import classNames from 'classnames'
import { Link } from 'react-router-dom'

import portfolio from 'redux/reducers/portfolio'

import './header.sass'

const renderLink = ({ name, link }, key) => (
  <li key={key}>
    <Link className="header__list-link" to={link}>
      {name}
    </Link>
  </li>
)

renderLink.propTypes = {
  name: PropTypes.string.isRequired,
  link: PropTypes.string.isRequired,
}

class Header extends Component {
  static propTypes = {
    headerLinks: PropTypes.array.isRequired,
  }

  state = {
    menuVisible: false,
  }

  toogleMenu = () => {
    this.setState(prevState => ({
      menuVisible: !prevState.menuVisible,
    }))
  }

  render() {
    const { menuVisible } = this.state
    const { headerLinks } = this.props

    return (
      <header className="header">
        <div className="header-wrapper">
          <a href="/" className="header__home">
            Selleo
          </a>
          <span
            role="button"
            tabIndex="0"
            className={classNames('header__menu', 'icon-menu', {
              'header__menu--active': !menuVisible,
            })}
            onClick={this.toogleMenu}
          />
        </div>
        <ul
          className={classNames('header__list', {
            'header__list--hidden': !menuVisible,
          })}
        >
          {_.map(headerLinks, (item, key) => renderLink(item, key))}
        </ul>
      </header>
    )
  }
}

const mapStateToProps = () => portfolio
export default connect(mapStateToProps)(Header)
