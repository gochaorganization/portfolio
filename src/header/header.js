import React from "react";

import "./header.sass";

const Header = () => (
  <header className="header">
    <a href="/" className="header__link">
      Selleo
    </a>
    <ul className="header__list">
      <li>
        <a className="header__list-link" href="#blog">
          Blog
        </a>
      </li>
      <li>
        <a className="header__list-link" href="#guide">
          Style Guide
        </a>
      </li>
      <li>
        <a className="header__list-link" href="#manuals">
          Manuals
        </a>
      </li>
      <li>
        <a className="header__list-link" href="#pages">
          Pages
        </a>
      </li>
      <li>
        <a className="header__list-link" href="#contacts">
          Contacts
        </a>
      </li>
    </ul>
  </header>
);

export default Header;
