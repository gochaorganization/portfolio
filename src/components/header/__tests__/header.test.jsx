import React from 'react'
import { shallow } from 'enzyme'
import { Header } from '../header'

describe('Header tests', () => {
  const props = {
    headerLinks: [
      { name: 'Blog', link: '#blog' },
      { name: 'Style Guide', link: '#guide' },
      { name: 'Manuals', link: '#manuals' },
      { name: 'Pages', link: '#pages' },
      { name: 'Contacts', link: '#contacts' },
    ],
  }

  test('renders correctly', () => {
    const wrapper = shallow(<Header {...props} />)
    expect(wrapper).toMatchSnapshot()
  })

  test('menu should render links', () => {
    const header = shallow(<Header {...props} />)
    expect(header.find('.header__list-link')).toHaveLength(5)
  })

  test('menu should be hidden', () => {
    const wrapper = shallow(<Header {...props} />)

    expect(wrapper.find('.header__list--hidden')).toHaveLength(1)
  })

  test('menu should be shown', () => {
    const wrapper = shallow(<Header {...props} />)
    wrapper.find('.header__menu').simulate('click')

    expect(wrapper.find('.header__list--hidden')).toHaveLength(0)
  })

  test('state menuVisible should change', () => {
    const wrapper = shallow(<Header {...props} />)
    wrapper.find('.header__menu').simulate('click')

    expect(wrapper.state().menuVisible).toBeTruthy()
  })

  //   test('dataTypes matches propTypes', () => {
  //     const wrapper = shallow(<Header {...props} />)
  //     wrapper.props.headerLinks.toBeArray()
  //   })
})

/*
1. Snapshot
2. Testing props. 
3. Testing data types (optional) - jest-extended 
4. Event Testing
5. Testing state

6.Check test coverage
*/
