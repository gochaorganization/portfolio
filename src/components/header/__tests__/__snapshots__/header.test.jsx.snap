// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header tests renders correctly 1`] = `ShallowWrapper {}`;
